src/EditorFeatures/Core/Tagging/TaggerContext.cs

```diff
@@ -86,15 +86,9 @@ public void ClearTags()
         public void SetSpansTagged(ImmutableArray<SnapshotSpan> spansTagged)
             => _spansTagged = spansTagged;
 
-        public IEnumerable<ITagSpan<TTag>> GetExistingContainingTags(SnapshotPoint point)
-        {
-            if (_existingTags != null && _existingTags.TryGetValue(point.Snapshot.TextBuffer, out var tree))
-            {
-                return tree.GetIntersectingSpans(new SnapshotSpan(point.Snapshot, new Span(point, 0)))
-                           .Where(s => s.Span.Contains(point));
```

Comment:
we were creating an entire list and then querying over it, when we can just ask teh tree directly (Without allocations) if it has any matches for a particular position